Rename "index type" to "address type" across the spec

bvisness opened the pull request:
Resolves #67.

Currently draft because I have not yet updated the spec interpreter, but the entire spec (core and JS) should be correct.

bvisness committed:
Rename "index type" to "address type" across the spec

rossberg commented on document/core/valid/types.rst:
```diff
@@ -501,22 +501,22 @@ Limits
 
-:math:`\idxtype~\limits~\reftype`
+:math:`\addrtype~\limits~\reftype`
 .................................
```
You'll have to adjust the underline here and in a few other places, otherwise Sphinx complains on occasion.

rossberg reviewed (approved):
LGTM % comment

bvisness committed:
Update "index type" to "address type" in the spec interpreter

bvisness committed:
Update section underlines

It is very important that you have precisely the correct number of dots.
It would be too much work for the computer otherwise.

bvisness commented:
Re-requested review now that the spec interpreter has also been updated.

rossberg reviewed (approved):
LGTM, with perhaps one request: in the interpreter, shorten the type name to addr_type to more closely match the spec's naming.

bvisness committed:
Rename "address type" to "addr type" in the spec interpreter

bvisness committed:
Revert one stray interpreter update

backes commented on document/js-api/index.bs:
```diff
@@ -873,18 +873,18 @@ enum TableKind {
-  undefined set(IndexValue index, optional any value);
-  readonly attribute IndexValue length;
+  AddressValue grow(AddressValue delta, optional any value);
+  any get(AddressValue index);
```
Should the parameter name have been updated to `address`, for `get` and `set`?

bvisness replied:
Ah, right you are. I'll make a follow-up PR.

bvisness replied:
Well, actually, this is arguably fine since the current spec uses "index" (predating memory64) and it's quite clear in context that this is referring to the index of an element. We can rename if we want, but I'm actually somewhat inclined to keep it as is. I'll open an issue just for any final bikeshedding.
